typescript/sdk/src/token/app.ts

```diff
@@ -0,0 +1,35 @@
+import { TokenRouter } from '@hyperlane-xyz/core';
+import { objKeys } from '@hyperlane-xyz/utils';
+
+import { appFromAddressesMapHelper } from '../contracts/contracts';
+import { HyperlaneAddressesMap, HyperlaneContracts } from '../contracts/types';
+import { MultiProvider } from '../providers/MultiProvider';
+import { GasRouterApp } from '../router/RouterApps';
+
+import { HypERC20Factories, hypERC20factories } from './contracts';
+
+export class HypERC20App extends GasRouterApp<HypERC20Factories, TokenRouter> {
+  constructor(contractsMap: any, multiProvider: any) {
+    super(contractsMap, multiProvider);
+  }
+
+  router(contracts: HyperlaneContracts<HypERC20Factories>): any {
+    for (const key of objKeys(hypERC20factories)) {
+      if (contracts[key]) {
+        return contracts[key];
+      }
+    }
+  }
```

Comment:
I worry a bit this will break down if eg there is an ISM or hook or something in the artifacts

Reply:
it looks for the specific type, how does ISM or hook change it?